updateengine/client_test.go

```diff
@@ -93,6 +93,20 @@ func TestRebootNeededSignal(t *testing.T) {
 		t.Fatal("RebootNeededSignal stopped prematurely")
 	}
 
+	c.ch <- nil
+
+	time.Sleep(10 * time.Millisecond)
+
+	select {
+	case stat := <-r:
```

Comment:
Do we want to log anything here? Otherwise the error is just eaten?

Reply:
Is this logging for the test only? It would better to log something for the test case of sending the `nil` signal. (right now `"some other ignored signal"` gets a status printed for it when running `go test` in locksmith/updateengine).

I agree on not logging `RebootNeededSignal` in the case that the loop `continue`s as this would pollute the shutdown logs. Logging one time only if `continue` happens by means of a counter or flag in the `for` loop would be OK though since `nil` signals don't always happen, and it might be useful to know from the shutdown logs if it did. We could add giving one log message for the first failure of `TryNewStatus` in, WDYT?

On logging for the test (in `client_test.go` a log should be added to indicate that the `nil` signal test is happening. I can add this in a bit.

Reply:
sgtm, and yeah I meant to comment on the continue in the code, not on this test.

Reply:
Sounds good, thanks.

On `client_test.go`: Had a second thought (from my comment above) on logging in `client_test.go`, and it is OK as it is since `t.Fatal` is called in case of failure on the `nil` signal test. So `client_test.go` will be kept as in this PR, unless there is anything else to add.

On `RebootNeededSignal`: We can still add a log for the first failure only (so logging the failure happens only once) of `TryNewStatus` in `RebootNeededSignal`, if we want to log *something* for the `nil` signal showing without polluting logs. However it may just be adding unnecessary logic to the code.

If all is good as the PR is now, I think it is OK to merge.